flink-java/src/main/java/org/apache/flink/api/java/ExecutionEnvironment.java

```diff
@@ -828,6 +828,18 @@ public JobExecutionResult execute() throws Exception {
 	 *                   be contacted to retrieve information relevant to the execution planning.
 	 */
 	public abstract String getExecutionPlan() throws Exception;
+
+	/**
+	 * Creates the AST and plan with which the system will execute the program, and return them as
+	 * sql-explanation format of HIVE.
+	 * Note that this needs to be called,  before the plan is executed.
+	 * 
+	 * @param extended The sql explanation mode.
+	 * @return The execution plan and AST of the program.
+	 * @throws Exception Thrown, if the compiler could not be instantiated, or the master could not
+	 *                   be contacted to retrieve information relevant to the execution planning.
+	 */
+	public abstract String getSqlExecutionPlan(boolean extended) throws Exception;
```

Comment:
I don't think we should add this method to all `ExecutionEnvironments`.
The JSON string returned by `getExecutionPlan()` should contain all relevant information. How about, we query the JSON document and parse the String in the Table API to reformat the information according to HIVE's explain visualization?